packages/expo-modules-core/android/src/test/java/expo/modules/kotlin/methods/SuspendMethodTest.kt

```diff
@@ -0,0 +1,156 @@
+package expo.modules.kotlin.methods
+
+import com.facebook.react.bridge.JavaOnlyArray
+import com.google.common.truth.Truth
+import expo.modules.PromiseMock
+import expo.modules.PromiseState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SuspendMethodTest {
+  @Test
+  fun `suspend block should resolve promise on finish`() = runTest {
+    val suspendMethod = SuspendMethod("test", emptyArray(), lazy { this }) {
+      delay(2000)
+    }
+    val promiseMock = PromiseMock()
+
+    suspendMethod.call(JavaOnlyArray(), promiseMock)
+    testScheduler.advanceTimeBy(3000)
+
+    Truth.assertThat(promiseMock.state).isEqualTo(PromiseState.RESOLVED)
+  }
+
+  @Test
+  fun `suspend block should reject promise when throws`() = runTest {
+    val suspendMethod = SuspendMethod("test", emptyArray(), lazy { this }) {
+      delay(2000)
+      throw IllegalStateException()
+    }
+    val promiseMock = PromiseMock()
+
+    suspendMethod.call(JavaOnlyArray(), promiseMock)
+    testScheduler.advanceTimeBy(3000)
+
+    Truth.assertThat(promiseMock.state).isEqualTo(PromiseState.REJECTED)
+  }
+
+  @Test
+  fun `suspend block should be cancelable`() {
+    val testScope = TestScope()
```

Comment:
I think each test should use the same type of scope, which is configured similarly to the module coroutine scope (`SupervisorJob`) even if it's not necessary for a specific test. This helps avoid possible situations when you modify the test in the future and it starts behaving differently than the actual module scope (or other tests). 

Maybe it would be worth doing
```kt
private lateinit var testScope: TestScope

@Before
fun setUp() {
  testScope = TestScope(SupervisorJob())
}
```